Index aliasing (#3861)
SumoLogic/sumologic-documentation: blog-service/2024-03-27-search.md

---
title: Index Aliasing for Search Queries
image: https://www.sumologic.com/img/logo.svg
keywords:
  - search
  - partitions
  - index
  - aliasing
hide_table_of_contents: true
authors:
  - url: https://help.sumologic.com/release-notes-service/rss.xml
    image_url: /img/release-notes/rss-orange.png
---

> We're excited to introduce the ability to index aliasing. With this update, you can now use an alias to point to one or more system indexes in the source expression of your search queries.

Reviewer: ```suggestion
We're excited to introduce the ability to index aliasing. With this update, you can now use an alias to point to one or more system indexes in the source expression of your search queries.
```
```suggestion
We're excited to introduce the ability to perform index aliasing. With this update, you can now use an alias to point to one or more system indexes in the source expression of your search queries.
```

> We're excited to introduce the ability to index aliasing. With this update, you can now use an alias to point to one or more system indexes in the source expression of your search queries.

Reviewer: ```suggestion
We're excited to introduce the ability to create index aliasing. With this update, you can now use an alias to point to one or more system indexes in the source expression of your search queries.
```

Reviewer 2: I don't think create would be the right word here, as user doesn't create anything on their end and rather references alias for system index in the query

Author: @atishya-22 , how about _We're excited to introduce index aliasing for Sumo Logic indexes._

Reviewer 2: "Index aliasing" in itself doesn't seem intuitive enough i believe. 

We're excited to introduce ability to reference sumo logic indexes with an alias. 

@kimsauce  Something similar to this? ^^

Author: sounds good - I'll apply it

Author: @atishya-22 how about: _We're excited to introduce the ability to reference any Sumo Logic index with an alias._

Reviewer 2: Sounds good. 
_We're excited to introduce the ability to reference Sumo Logic system indexes with an alias._

seems to give more precise idea?

Author: perfect!

> We're excited to introduce the ability to index aliasing. With this update, you can now use an alias to point to one or more system indexes in the source expression of your search queries.

Author: ```suggestion
We're excited to introduce the ability to reference any Sumo Logic index with an alias. With this update, you can now use an alias to point to one or more system indexes in the source expression of your search queries.
```

> We're excited to introduce the ability to reference any Sumo Logic index with an alias. With this update, you can now use an alias to point to one or more system indexes in the source expression of your search queries.

Author: ```suggestion
We're excited to introduce the ability to reference Sumo Logic system indexes with an alias. With this update, you can now use an alias to point to one or more system indexes in the source expression of your search queries.
```

### How it works

> With index aliasing, you can create an alias to point to one or more system indexes such as [`sumologic_default`](/docs/manage/partitions/run-search-against-partition/#search-the-default-partition) in the source expression of your query. Both the operator part and results will consist of actual index names.

Author: @atishya-22 , thanks for clarifying. It looks like you've updated this paragraph. Is your new wording ok?

> With index aliasing, you can create an alias to point to one or more system indexes such as [`sumologic_default`](/docs/manage/partitions/run-search-against-partition/#search-the-default-partition) in the source expression of your query. Both the operator part and results will consist of actual index names.

Reviewer 2: yes, that seems fine to me now. have also applied the suggestion you've commented.

> With index aliasing, you can use an alias to point to one or more system indexes such as [`sumologic_default`](/docs/manage/partitions/run-search-against-partition/#search-the-default-partition) in the source expression of your query. Both the operator part and results of query will consist of actual index names.

Author: ```suggestion
With index aliasing, you can use an alias to point to one or more system indexes, such as [`sumologic_default`](/docs/manage/partitions/run-search-against-partition/#search-the-default-partition) in the source expression of your query. Both the operator part and results of your query will consist of actual index names. 
```

### How it works

With index aliasing, you can create an alias to point to one or more system indexes such as [`sumologic_default`](/docs/manage/partitions/run-search-against-partition/#search-the-default-partition) in the source expression of your query. Both the operator part and results will consist of actual index names.

In addition to `sumologic_default`, we have several other Sumo Logic-defined system indexes. As a shortcut, rather than prefacing `sumologic_` when referencing system indexes in a search, you can alias these indexes by typing an underscore at the beginning. For example, `sumologic_default` and `_default` will return the same results. 

Learn more [here](/docs/manage/partitions/run-search-against-partition#index-aliasing).

### What you need to know

* Effective with this update, you can no longer create your own indexes (user-created, non-system indexes), you cannot lead with an underscore (`_`). This will be reserved only for aliasing system indexes. All system partitions that have been created with `_<index-name>` are transitioned to `sumologic_<index-name>`.

> * Sumo Logic security indexes will no longer start with `sumologic_`. Examples of security indexes include [Cloud SIEM security](/docs/cse/records-signals-entities-insights/search-cse-records-in-sumo#partitions-for-cse-records) indexes, [Audit](/docs/manage/security/audit-indexes/audit-index/) indexes, and [Root Cause Explorer](/docs/observability/root-cause-explorer) indexes. For all other Sumo Logic-created indexes, aliasing is permitted.

Reviewer 2: I think this got wrongly perceived from the requirements doc. We're not changing anything with the naming of current indexes. We're only introducing aliasing to allow a shorter name.
